A11y colors

hfcRed opened the pull request:
This PR adds a table to the custom colors setting to show the contrast ratio of the selected colors.

Additionally I also added the ability to preview the colors as you're setting them, as well as fixing the reset button to reflect the change in the UI.

Added a bit of styling on the tables as well. Tested on all 3 major browsers + mobile.

https://github.com/user-attachments/assets/ebb0e2b8-13ce-43ba-91a5-7ecf38d802be

I do wonder if we would want to block the submit unless at least AA is met on all combinations, or if this should just be a honor system 🤔

Sendouc reviewed (commented):
Thank you for raising this topic up 🙌 Still some things to figure out. Left some comments for discussion.

I also have some concerns with the design. The element now takes a huge amount of space on the page:

<img width="260" alt="image" src="https://github.com/user-attachments/assets/36f2da9f-e45c-44c5-bcca-a61ce1978006" />

Could we for example put the contrast checker behind a tab or something (tab directly above the current color input) so it takes the same amount of space?

Did you test this for team page custom color inputs? It seems there is some horizontal scroll there

<img width="329" alt="image" src="https://github.com/user-attachments/assets/9531aa45-b62d-4194-9afd-6b59710f4a66" />

hfcRed commented on app/components/CustomizedColorsInput.tsx:
```diff
@@ -65,6 +116,166 @@ export function CustomizedColorsInput({
+						<React.Fragment key={contrast.colors.join("-")}>
+							<div>{t(`custom.colors.${contrast.colors[0]}`)}</div>
+							<div>{t(`custom.colors.${contrast.colors[1]}`)}</div>
+							<div
+								className={`colors__contrast ${
+									contrast.contrast.AA.failed ? "fail" : "success"
+								}`}
+							>
+								{contrast.contrast.AA.ratio}
+							</div>
+							<div
+								className={`colors__contrast ${
+									contrast.contrast.AAA.failed ? "fail" : "success"
+								}`}
+							>
```
I tried to match the markup of the custom color settings, but I dont mind changing it to be an actual table instead. I considered adding icons or text for fail/pass, but horizontal space is already very tight here. Are there already icons for this available in the codebase?

hfcRed commented on app/components/CustomizedColorsInput.tsx:
```diff
@@ -65,6 +116,166 @@ export function CustomizedColorsInput({
+				<div>AAA</div>
+				{contrasts.map((contrast) => {
+					return (
+						<React.Fragment key={contrast.colors.join("-")}>
```
I just join the two color names that are being compared. Since every pair only exists once, it results in a unique ID.

hfcRed commented on app/features/team/routes/t.$customUrl.edit.tsx:
```diff
@@ -78,7 +78,7 @@ export default function EditTeamPage() {
 				</FormWithConfirm>
 			) : null}
-			<Form method="post" className="stack md items-start">
+			<Form method="post" className="stack md items-start" reloadDocument>
```
This is a workaround for the live color preview since the unmount runs AFTER the custom CSS loads after a submit. Here's what happens without the document reload:

https://github.com/user-attachments/assets/d4d26452-b8c3-436a-9186-59ec19b3abec

Reloading the page is the most consistent thing I could come up with unfortunately, and I also don't want to drop the live preview as imo its a big QoL improvement. Maybe you have an idea on how to better handle the unmount.

hfcRed replied:
The horizontal scroll is intentional! There is just not enough vertical space available, especially on mobile and if we want to add fail/pass icons/text as well. So I opted to style the tables to retain their content size instead of using ``fr`` units, then let it scroll if it runs out of space.

I can split the content into tabs for sure tho, shouldnt be much of a problem. Do you want all of this to stay in a single component still?

hfcRed commented:
The horizontal scroll is intentional! There is just not enough vertical space available, especially on mobile and if we want to add fail/pass icons/text as well. So I opted to style the tables to retain their content size instead of using fr units, then let it scroll if it runs out of space.

I can split the content into tabs for sure tho, shouldnt be much of a problem. Do you want all of this to stay in a single component still?

hfcRed commented on app/components/CustomizedColorsInput.tsx:
```diff
@@ -65,6 +116,166 @@ export function CustomizedColorsInput({
+							<div
+								className={`colors__contrast ${
+									contrast.contrast.AA.failed ? "fail" : "success"
+								}`}
```
Good point, fixed in latest commit

Sendouc commented:
> The horizontal scroll is intentional! There is just not enough vertical space available, especially on mobile and if we want to add fail/pass icons/text as well. So I opted to style the tables to retain their content size instead of using fr units, then let it scroll if it runs out of space. This prevents awkward wrapping and squeezing.

👍 

> I can split the content into tabs for sure tho, shouldnt be much of a problem. Do you want all of this to stay in a single component still?

You can divide them to components however you think is the best :) I would use the same tabs component as the front page has for showcase tournaments

hfcRed commented:
* Switched to table element
* Moved description into popover
* Added icons for fail/pass

![chrome_nxGeEq3O7V](https://github.com/user-attachments/assets/4ada29fa-f4e5-498c-83ea-0755a89a2be8)

hfcRed commented:
Implemented suggestion as per
https://discord.com/channels/299182152161951744/784073459516964954/1373698676610043954

Was able to remove the reloadDocument

Sendouc commented:
Nice job getting rid of the `reloadDocument`.

Note: InfoPopover styles were broken, now fixed in rewrite branch.

I would still think tabs make sense. There are people who are supporters who don't care about the custom colors at all for example. I still think that even after the rework it is taking a lot of vertical space and people have to scroll past those to get to the parts they actually want to update.

hfcRed commented:
I tried using tabs, but constantly switching between tabs is very tedious, especially when finetuning colors. Would a ``<details>`` work maybe? Then the entirety of the custom colors section can be collapsed

Sendouc commented:
Yeah details sounds good. Then only thing design wise I noticed is that pre tag makes the popover text look a bit off as it is not used anywhere else on the site.

The live updating is so satisfying haha

Sendouc commented:
Awesome work 🚀
